### PR TITLE
pop-up: save dirty files before packing/launching

### LINKED ISSUE #1712: Warn about un-saved files when packing/installing mod
When the user has un-saved changes to their files when trying to install or pack the mod, Wolvenkit should show them a popup (Save now / pack anyway)

### PR DESCRIPTION
**Pull request by manavortex**
# save dirty files before packing/launching

**Additional notes:**
closes #1712

### CODE REVIEW

File: WolvenKit.App/ViewModels/Documents/DocumentViewModel.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public DocumentViewModel(string path) : base(Path.GetFileName(path), path)
     public bool IsDirty
     {
         get => _isDirty;
-        protected set => SetProperty(ref _isDirty, value);
+        set => SetProperty(ref _isDirty, value);
```
Comment (seberoth) [inline WolvenKit.App/ViewModels/Documents/DocumentViewModel.cs:37]: Why did you make it public? Should kept private/protected.

**Review by seberoth — CHANGES_REQUESTED** [review]

File: WolvenKit.App/ViewModels/Documents/DocumentViewModel.cs (hunk 1 above)
Comment (manavortex) [reply WolvenKit.App/ViewModels/Documents/DocumentViewModel.cs:37]: The compiler didn't let me and insisted it needed to be public. It's protected in the interface, though.

Comment (seberoth) [reply WolvenKit.App/ViewModels/Documents/DocumentViewModel.cs:37]: Remove the setter from the interface

Comment (manavortex) [reply WolvenKit.App/ViewModels/Documents/DocumentViewModel.cs]: Ah, yeah, makes sense. Fixed

**Review by seberoth — APPROVED** [review]

**Pull request merged**